floors/floors.go

```diff
@@ -0,0 +1,135 @@
+package floors
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+
+	"github.com/mxmCherry/openrtb/v16/openrtb2"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/currency"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const (
+	DEFAULT_DELIMITER      string = "|"
+	CATCH_ALL              string = "*"
+	SKIP_RATE_MIN          int    = 0
+	SKIP_RATE_MAX          int    = 100
+	MODEL_WEIGHT_MAX_VALUE int    = 100
+	MODEL_WEIGHT_MIN_VALUE int    = 0
+	ENFORCE_RATE_MIN       int    = 0
+	ENFORCE_RATE_MAX       int    = 100
+)
+
+type FloorConfig struct {
+	FloorEnabled      bool
+	EnforceRate       int
+	EnforceDealFloors bool
+}
+
+func NewFloorConfig(priceFloor config.PriceFloors) *FloorConfig {
+
+	floorConfig := FloorConfig{
+		FloorEnabled:      priceFloor.Enabled,
+		EnforceRate:       priceFloor.EnforceFloorsRate,
+		EnforceDealFloors: priceFloor.EnforceDealFloors,
+	}
+
+	return &floorConfig
+}
+
+func (fc *FloorConfig) Enabled() bool {
+	return fc.FloorEnabled
+}
+
+func (fc *FloorConfig) GetEnforceRate() int {
+	return fc.EnforceRate
+}
+
+func (fc *FloorConfig) EnforceDealFloor() bool {
+	return fc.EnforceDealFloors
+}
+
+type Floor interface {
+	Enabled() bool
+	GetEnforceRate() int
+	EnforceDealFloor() bool
+}
+
+// IsRequestEnabledWithFloor will check if floors is enabled in request
+func IsRequestEnabledWithFloor(Floors *openrtb_ext.PriceFloorRules) bool {
+
+	if Floors != nil && Floors.Enabled != nil && !*Floors.Enabled {
+		return *Floors.Enabled
+	}
+
+	return true
+}
+
+// UpdateImpsWithFloors will validate floor rules, based on request and rules prepares various combinations
+// to match with floor rules and selects appripariate floor rule and update imp.bidfloor and imp.bidfloorcur
+func UpdateImpsWithFloors(floorExt *openrtb_ext.PriceFloorRules, request *openrtb2.BidRequest, conversions currency.Conversions) []error {
```

Comment:
Nitpick: Might be better named `ModifyImpsWithFloors` to make it absolutely clear it's modifying input.

Reply:
updated to ModifyImpsWithFloors()